crypto/types/types.go

```diff
@@ -0,0 +1,24 @@
+package types
+
+import (
+	proto "github.com/gogo/protobuf/proto"
+	tmcrypto "github.com/tendermint/tendermint/crypto"
+)
+
+// PubKey interface extends proto.Message
+// and tendermint crypto.PubKey
+type PubKey interface {
+	proto.Message
+	tmcrypto.PubKey
+}
+
+// PrivKey interface extends proto.Message
+// and tendermint crypto.PrivKey
+type PrivKey interface {
+	proto.Message
+	tmcrypto.PrivKey
```

Comment:
`tmcrypto.PrivKey` has a `PubKey()` method, which returns a `tmcrypto.PubKey`. I believe it should return a `PubKey` (ours) instead. I'm running into some problems related to `privKey.PubKey()` not being a PubKey in my refactoring.

This

Reply:
cf. https://github.com/cosmos/cosmos-sdk/pull/7147#discussion_r487395924